x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/DriverRunner.java

```diff
@@ -87,4 +104,23 @@ private void done() {
             start(driver, driverListener);
         }
     }
+
+    private void mergeResponseHeaders(AtomicArray<Map<String, List<String>>> responseHeaders) {
+        final Map<String, Set<String>> merged = new HashMap<>();
```

Comment:
I propose a TreeMap to have the headers in order.

Reply:
Thanks Costin. But it doesn't matter much since we will lose this order at https://github.com/elastic/elasticsearch/blob/c3f164a863870a91faeb85d4ff34116d4ed2cb89/server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java#L789